File: problem20.js
function factorial(n) {
  var fact = 1;
  for (var i = 1; i<=n; i++) {
    fact = fact * i;
  }
  return fact;
}

function findSum(n) {

  var sum = 0;
  while ( n > 0 ) {
    sum = sum + n%10;
    n = parseInt(n / 10);
  }
  return sum;

}

function convertNumToArray(n) {

  var num = [];
  while ( n > 0 ) {
    num.push(n%10);
    n = parseInt(n / 10);
  }
  return num;

}

function sumFactorialDigits(n) {

  if ( n <= 10 ) {
    return findSum(factorial(n));
  }

  var fact = convertNumToArray(factorial(10));
  //console.log("Factorial of 10 is "+fact);

  var temp = 0;
  var carryFwd = 0;
  for (var i = 11; i <= n; i++) {
    for ( var j = 0; j < fact.length; j++) {
      temp = carryFwd + fact[j]*i;
      fact[j] = temp % 10;
      carryFwd = parseInt(temp / 10 );
    }
    
    var carryFwdInArr = [];
    if ( carryFwd > 0 ) {
      carryFwdInArr = convertNumToArray(carryFwd);
      for ( var x = 0; x < carryFwdInArr.length; x++ ) {
        fact[j] = carryFwdInArr[x]; 
        j++;
      }
      carryFwd = 0;
    }
    //console.log("Factorial of "+i+" is "+fact);
  }

  //console.log("Factorial of "+i+" is "+fact);
  var ans = 0;
  for (var i=0; i < fact.length; i++) {
    ans = ans + fact[i];
  }
  return ans;
  
}

sumFactorialDigits(25);

File: go/problem20.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(FactorialDigitSum(10))
}

func FactorialDigitSum(n int) int {

	if n <= 10 {
		return findSum(factorial(n))
	}

	fact := convertNumToArray(factorial(10))

	temp := 0
	carryFwd := 0
	for i := 11; i <= n; i++ {
		for j := 0; j < len(fact); j++ {
			temp = carryFwd + fact[j]*i
			fact[j] = temp % 10
			carryFwd = temp / 10
		}

		var carryFwdInArr []int
		if carryFwd > 0 {
			carryFwdInArr = convertNumToArray(carryFwd)
			for x := 0; x < len(carryFwdInArr); x++ {
				fact = append(fact, carryFwdInArr[x])
			}
			carryFwd = 0
		}

	}

	ans := 0
	for i := 0; i < len(fact); i++ {
		ans += fact[i]
	}
	return ans
}

func factorial(n int) int {

	fact := 2
	for i := 3; i <= n; i++ {
		fact = fact * i
	}
	return fact
}

func findSum(n int) int {

	sum := 0
	for n > 0 {
		sum = sum + n%10
		n = n / 10
	}
	return sum

}

func convertNumToArray(n int) []int {

	var num []int
	for n > 0 {
		num = append(num, n%10)
		n = n / 10
	}
	return num

}
